timesketch/ui/js/directives.js

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var directives = angular.module('timesketch.directives', []);
+
+directives.directive("butterbar", function() {
+    return {
+        restrict : "A",
+        link : function(scope, element, attrs) {
+            scope.$on("httpreq-start", function(e) {
+                element.css({"display": "block"});
+            });
+
+            scope.$on("httpreq-complete", function(e) {
+                element.css({"display": "none"});
+            });
+        }
+    };
+});
+
+directives.directive('timelineColor', function () {
+    return {
+      restrict: 'A',
+      link: function (scope, elem, attrs) {
+          var timeline_colors = scope.meta.timeline_colors;
+          elem.css("background", "#" + timeline_colors[scope.event._index])
+      }
+    }
+});
+
+directives.directive('timelineName', function () {
+    return {
+      restrict: 'A',
+      link: function (scope, elem, attrs) {
+          var timeline_names = scope.meta.timeline_names;
+          elem.addClass("label");
+          elem.css("color", "#999");
+          elem.append(timeline_names[scope.event._index])
+      }
+    }
+});
+
+directives.directive('indexChooser', function() {
```

Comment:
I'd suggest having an app-specific prefix for all timesketch controllers/services/directives registered in global AngularJs namespace. Something like "ts" (i.e. tsIndexChooser).